platform/darwin/src/MGLAttributedExpression.h

```diff
@@ -0,0 +1,23 @@
+#import "MGLFoundation.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NSString * MGLAttributedExpressionKey NS_EXTENSIBLE_STRING_ENUM;
+
+FOUNDATION_EXTERN MGL_EXPORT MGLAttributedExpressionKey const MGLFontNamesAttribute;
+FOUNDATION_EXTERN MGL_EXPORT MGLAttributedExpressionKey const MGLFontSizeAttribute;
+
+MGL_EXPORT
+@interface MGLAttributedExpression : NSObject
```

Comment:
I'm using the same mental model between `NSString` and `NSAttributedString` where you can't assign the later to the first one and vice versa. I think `MGLAttributedExpression` keeps the concept of an object that contains an expression with attributes.

Reply:
This is a little different though: NSString doesn’t hold an array of NSAttributedStrings.

Reply:
I agree with @1ec5. How about something like `MGLTextFormatAttributes`? The `MGLText` prefix already exists as part of MGLSymbolStyleLayer...

Reply:
yes, we have:

```
textField(
         format(
           formatEntry("this is", formatFontScale(0.75)),
           formatEntry(
             concat(literal("\n"), get(TITLE_FEATURE_PROPERTY)),
             formatFontScale(1.25),
             formatTextFont(new String[] {"DIN Offc Pro Bold", "Arial Unicode MS Regular"})
           )
         )
       ),
```

Reply:
I think the concept that resembles the intent is `MGLAttributedExpression`, in regards to `MGLTextFormatAttributes` the underlaying object is not text per se (although it may contain a string constant).